src/wemmick/grpc_server.py

```diff
@@ -0,0 +1,70 @@
+from concurrent import futures
+import grpc
+from grpc_reflection.v1alpha import reflection
+import wemmick_pb2
+import wemmick_pb2_grpc
+import wemmick.api
+
+
+class Wemmick(wemmick_pb2_grpc.WemmickServicer):
+
+    def HelloWorld(self, request, context):
+        return wemmick_pb2.Response(message='Hello World!')
+
+    def ListDataSources(self, request, context):
+        data_sources = wemmick.api.list_datasources()
+        result = []
+        for data_source_item in data_sources:
+            credentials = wemmick_pb2.DataSource.Credentials(
+                connection_string=data_source_item['credentials']['connection_string']
+            )
+            data_source = wemmick_pb2.DataSource(
+                name=data_source_item['name'],
+                class_name=data_source_item['class_name'],
+                module_name=data_source_item['module_name'],
+                credentials=credentials
+            )
+            result.append(data_source)
+
+        return wemmick_pb2.DataSourceList(data_sources=result)
+
+    def ListExpectationSuites(self, request, context):
+        suites = wemmick.api.list_expectation_suites()
+        result = []
+        for suite_item in suites:
+            result.append(suite_item.expectation_suite_name)
+
+        return wemmick_pb2.ExpectationSuitesList(expectation_suites=result)
+
+    def CreateExpectationSuiteFromJsonSchema(self, request, context):
+        create_suite = wemmick.api.CreateExpectationSuiteFromJsonSchema(
+            file_path=request.file_path,
+            suite_name=request.suite_name
+        )
+        create_suite.run()
+        return wemmick_pb2.Response(message=f'Successfully created {request.suite_name} expectation suite')
+
+    def CreateExpectationSuiteFromAvroSchema(self, request, context):
+        create_suite = wemmick.api.CreateExpectationSuiteFromAvroSchema(
+            file_path=request.file_path,
+            suite_name=request.suite_name
+        )
+        create_suite.run()
+        return wemmick_pb2.Response(message=f'Successfully created {request.suite_name} expectation suite')
```

Comment:
I'm likely missing something about the intended use case here, and/or have not communicated the intent of wemmick (which is dual purpose - a server that can be triggered to run a validation and an interactive CLI tool to create suites):

My impression was that the grpc interface was for triggering validations, not for creation of suites which would require passing files around. From a UX perspective it's going to be way easier to use the wemmick CLI to create suites than to invoke wemmick via grpc and shuttles files around. I'd motion that we remove all but the validation functionality.

Reply:
My vision has been along the lines of Wemmick being a service hosted in your cloud provider which anyone in the team could create new suites from schemas, run validations with these suites, and view results. All without ever having to install even python on their local machine. So my scope for [751](https://bera.kanbanize.com/ctrl_board/9/cards/751/details/) has been to give cli equivalent functionality in gRPC

This allows to programatically create the latest expectation suites from a schema profiler before running a validation operator. So you would just have to point to any schema file and be able to validate a table against it.

Reply:
That makes sense to me. Thanks for the clarification.

Reply:
I think your both right on :D 

But the MVP the gRPC interface only needs to invoke existing suites and JSON Schema generated ones.

We can expand what the interface does in additional iterations which is why I want gRPC. I want the commands and types to be "discovered" by the CLI.


Also, the REST interface just needs to do the same invoking but via from Cloud Scheduler.